From 20d49d82d5f3ce2ee91fdb1156c9a34181fe03c1 Mon Sep 17 00:00:00 2001
From: Haiyue Wang <haiyuewa@163.com>
Date: Wed, 11 Mar 2026 03:20:52 +0800
Subject: [PATCH 17/18] PSUSensor: fix hwmon name mismatch

The commit [1] "Move datastructures to init before main" breaks the
hwmon name comparison.

The original implementation performs a case-insensitive comparison
on the 'type' field using I2CDeviceComparator with strcasecmp().

Change the comparison to use the 'name' field of I2CDeviceType instead,
for example:
    {"ADM1266", I2CDeviceType{"adm1266", true}}
                                ^

[1]: https://gerrit.openbmc.org/c/openbmc/dbus-sensors/+/60548

Change-Id: Idea493625b206a28395b2e6cdecadcdf9431f6f7
Signed-off-by: Haiyue Wang <haiyuewa@163.com>
---
 src/psu/PSUSensorMain.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/psu/PSUSensorMain.cpp b/src/psu/PSUSensorMain.cpp
index c08b6c3..1a58712 100644
--- a/src/psu/PSUSensorMain.cpp
+++ b/src/psu/PSUSensorMain.cpp
@@ -644,7 +644,7 @@ static void createSensorsCallback(
 
         if (std::ranges::find_if(sensorTypes.begin(), sensorTypes.end(),
                                  [pmbusName](const auto& a) {
-                                     return a.first == pmbusName;
+                                     return a.second.name == pmbusName;
                                  }) == sensorTypes.end())
         {
             // To avoid this error message, add your driver name to
-- 
2.52.0

